pkg/build/api/validation/validation.go

```diff
@@ -64,14 +64,15 @@ func ValidateBuildConfig(config *buildapi.BuildConfig) field.ErrorList {
 	fromRefs := map[string]struct{}{}
 	specPath := field.NewPath("spec")
 	triggersPath := specPath.Child("triggers")
+	buildFrom := buildutil.GetImageStreamForStrategy(config.Spec.Strategy)
```

Comment:
(i'm ok with canning this PR if it's a problem but i was trying to make @jwforres' life easier :) )